### PR TITLE
Gameboard

### PR DESCRIPTION
**Pull request by K-Rye**
Refactored Gameboard class by separating functionality into two separate functions. Added a shuffle feature and round update. Removed unnecessary comments and console.logs.  

@brittanystoroz @thatPamIAm

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -60,6 +60,7 @@ typings/
 # next.js build output
 .next
 
-
 .DS_Store
 
+.eslintrc
```
Comment (chris023) [inline .gitignore:65]: We want this to be in the repo so that styling is consistent across contributors

**Review by chris023 — APPROVED** [review]

File: js/Gameboard.js [hunk 2]
```diff
@@ -1,23 +1,52 @@
-class Gameboard {
+  class Gameboard {
+    constructor() {
+      this.possibleCategories = this.generateCategories();
+    }
+
+  
+    const generateRandomIndex = (length) => {
```
Comment (brittanydionigi) [inline js/Gameboard.js:7]: You don't want to be creating named arrow functions inside of a class -- you should be creating methods instead like so:

```js
generateRandomIndex() {
  // do stuff
},

chooseQuestion() {
  // do stuff
}
```

File: js/Gameboard.js [hunk 3]
```diff
@@ -1,23 +1,52 @@
-class Gameboard {
+  class Gameboard {
+    constructor() {
+      this.possibleCategories = this.generateCategories();
```
Comment (brittanydionigi) [inline js/Gameboard.js:3]: `this.generateCategories` doesn't seem to be a method that you've defined on this class. You would be getting an error upon instantiating a new Gameboard with this code in its current state.

File: js/Gameboard.js [hunk 4]
```diff
@@ -1,23 +1,52 @@
...
+      return randomQuestion;
     }
-    //return gameboard;
+
+    const shuffle = (a) => {
+      for (let i = a.length - 1; i > 0; i--) {
+        const j = Math.floor(Math.random() * (i + 1));
+        [a[i], a[j]] = [a[j], a[i]];
+      }
+      return a;
```
Comment (brittanydionigi) [inline js/Gameboard.js:29]: This code is really difficult to read, can you explain what's happening here line-by-line?

**Comment by brittanydionigi** [comment]
@K-Rye @chris023 Quite a bit of syntax here that seems like it would be broken. Take some time to review basic `class` syntax, as that's a fundamental concept we need to have under our belts before moving forwards.

**Pull request merged**